app/assets/javascripts/pages/agent-edit-page.js.coffee

```diff
@@ -16,10 +16,25 @@ class @AgentEditPage
     if $("#agent_type").length
       $("#agent_type").on "change", => @handleTypeChange(false)
       @handleTypeChange(true)
+      $('#agent_type').select2
+        width: 'resolve'
+        formatResult: formatAgentForSelect
+        escapeMarkup: (m) ->
+          m
+        matcher: (term, text, opt) ->
+          description = opt.attr('title')
+          text.toUpperCase().indexOf(term.toUpperCase()) >= 0 or description.toUpperCase().indexOf(term.toUpperCase()) >= 0
+
     else
       @enableDryRunButton()
       @buildAce()
 
+  formatAgentForSelect = (agent) ->
+    originalOption = agent.element
+    description = $(originalOption).attr('title')
+    description = if /^[a-zA-Z]/.test(description) then description else ''
+    '<strong>' + agent.text + '</strong><br/>' + description
```

Comment:
I think this will break if the Agent text / description has any html-like characters in it.  I'd suggest adding `underscore.js` to our `application.js.coffee` and then using `_.escape`.